simplified-app-shared/src/main/java/org/nypl/simplified/app/AnnotationsManager.kt

```diff
@@ -0,0 +1,600 @@
+
+      postAnnotation(requestUri, jsonBodyString, 20, { isSuccessful, response ->
+        if (isSuccessful) {
+          //TODO is there any reason that we need to save and update the annotation ID created for a reading-position upload?
+          LOG.debug("Success: Marked Reading Position To Server. Response: $response")
+        } else {
+          LOG.error("Annotation not posted.")
+        }
+      })
+
+    } catch (e: IOException) {
+      LOG.error("Error serializing JSON from Map Object")
+      return
+    }
+  }
+
+  private fun postAnnotation(uri: String, bodyParameters: String, timeout: Int?,
+                             completion: (isSuccessful: Boolean, annotationID: String?) -> Unit)
+  {
+    try {
+      val jsonObjectBody = JSONObject(bodyParameters)
+      val request = NYPLJsonObjectRequest(
+          Request.Method.POST,
+          uri,
+          credentials.barcode.toString(),
+          credentials.pin.toString(),
+          jsonObjectBody,
+          null,
+          Listener<JSONObject> { response ->
+            LOG.debug("Annotation POST: Success 200.")
+            //TODO am I doing anything with the returned annotation yet? maybe just for bookmarks..
+            val serverAnnotationID = response.getString("id")
+            completion(true, serverAnnotationID)
+          },
+          Response.ErrorListener { error ->
+            LOG.error("POST request fail! Network Error Cause: ${error.cause ?: "error cause was null"}")
+            completion(false, null)
+          })
+
+      if (timeout != null) {
+        request.retryPolicy = DefaultRetryPolicy(
+            timeout * 1000,
+            DefaultRetryPolicy.DEFAULT_MAX_RETRIES,
+            DefaultRetryPolicy.DEFAULT_BACKOFF_MULT
+        )
+      }
+
+      requestQueue.add(request)
+
+    } catch (e: java.lang.Exception) {
+      LOG.error("Post Annotation: Error creating JSONObject from Kotlin Map Type.")
+    }
+  }
+
+
+  //TODO WIP. Converted from Swift. Not yet tested.
+  /**
+   * Get a list of any bookmark-type annotations created for the given book and the current user.
+   * @param bookID Identifier for the entry
+   * @param uri The Annotation ID/URI for the OPDS Entry
+   * @param completion Called asynchronously by the network request to return the bookmarks
+   */
+  fun requestBookmarksFromServer(bookID: String?, uri: Uri?,
+                                 completion: (bookmarks: Any?) -> Unit) {
+
+    if (!syncIsPossibleAndPermitted()) {
+      LOG.debug("Account does not support sync or sync is disabled.")
+      completion(null)
+      return
+    }
+    if (bookID == null || uri == null) {
+      LOG.error("Required parameter was unexpectedly null")
+      return
+    }
+
+    NYPLJsonObjectRequest(
+        Request.Method.GET,
+        uri.toString(),
+        credentials.barcode.toString(),
+        credentials.pin.toString(),
+        null,
+        Listener<JSONObject> { response ->
+
+          try {
```

Comment:
This callback lambda is quite large, for readability I recommend making it a new method.